opencog/nlp/learn/make-disjuncts.scm

```diff
@@ -84,17 +85,133 @@
 
-;Applies the map function to create MST nodes for all the word pairs. It creates
-;nodes for only those word pairs which have a mutual information not equal to
-;-1000
-(define (create-MST-nodes text)
-	(map create-evaluation-link (filter criteria? (parse text)) ))
-
 ;This function is used by the higher-order-function filter to selectively apply
-;the create-evaluation-link function to only those word pairs which do not have 
-;a mutual information of -1000
+;the create-evaluation-link function to only those word pairs which do not 
+;have a mutual information of -1000.
 (define (criteria? wp)
 	(if (= -1000 (get-mutual-information wp))
 		#f
 		#t))
 
+;Applies the map function to create MST nodes for all the word pairs. It 
+;creates nodes for only those word pairs which have a mutual information not 
+;equal to -1000
+(define (create-MST-nodes text)
+	(map create-evaluation-link (filter criteria? (parse text)) ))
+
+;This function will give the name of the MSTNode which exists in a given 
+;Evaluation Link.
+(define (get-MST-name outset)
+	(cog-name (car outset)))
+
+;This function will return the first word present in the ListLink component of 
+;an EvaluationLink.
+(define (get-first-evaluation-word outset)
+	(cog-name (car (cog-outgoing-set (car (cdr outset))))))
+
+;This function will return the second word present in the ListLink component of  
+;an EvaluationLink.
+(define (get-second-evaluation-word outset)
+	(cog-name (car (cdr (cog-outgoing-set (car (cdr outset)))))))
+
+;This function will get all the unique words that were present in the sentence
+;for which the MST nodes were created.
+(define (get-sentence-words sentence)
+	(string-split sentence #\ ))
+
+;The extract-disjunct function returns a list of disjuncts for that particular 
+;word. However, that list contains many null lists also. These lists need to 
+;be eliminated before they can be processed further.
+(define (filter-disjunct disjuncts)
+	(filter filter-disjunct-criteria? disjuncts))
+
+;This is the criteria for the filter function which cleans up the disjuncts 
+;list for a list which does not contain any null lists.
+(define (filter-disjunct-criteria? x)
+	(not (null? x)))
+
+;This function returns the connector of the disjunct. Eg:- "MC+" will return MC
+(define (get-disjunct-connector disjunct)
+	(substring disjunct 0 (- (string-length disjunct) 1)))
+
+;This function returns the sign of the disjunct. Eg:- "MC+" will return +
+(define (get-disjunct-sign disjunct)
+	(substring disjunct (- (string-length disjunct) 1)))
+
+;This function takes as input a word and a list of MST nodes created by the 
+;create-MST-nodes function. For the given word, it gives the disjuncts of that 
+;word as found in the list of nodes give to it as input.
+(define (extract-disjunct word nodes)
+	(if (not (null? nodes))
+		(let* (
+				[node (car nodes)]
+				[outset (cog-outgoing-set node)]
+				[linkname (get-MST-name outset)]
+				[firstword (get-first-evaluation-word outset)]
```

Comment:
This seems to be the only place where `get-first-evaluation-word` is used (and is ever likely to be used)  Thus, could you move its definition to just after line 144? That way, `get-first-evaluation-word` will be scoped only inside of the `extract-disjunct` function, instead of being part of the global namespace.  Similar remarks apply for many of the other functions, especially the short ones.

The other way to hide locally-defined functions and keep them out of the global namespace is to use modules, (i.e. to define this file as a module) ... but, right now, in opencog, we do not yet have uniform module-coding standards, and module support is a bit shaky and possibly buggy.   I hope to clear that up by summertime.